internal/server/handlers_test.go

```diff
@@ -307,24 +307,42 @@ func TestAPI_DeleteProvider(t *testing.T) {
 }
 
 // withAdminIdentity may be used with setupServer to setup the server
-// with an admin machine identity and access key
+// with an admin identity and access key
 func withAdminIdentity(_ *testing.T, opts *Options) {
 	opts.Identities = append(opts.Identities, Identity{
 		Name:      "admin",
 		AccessKey: "BlgpvURSGF.NdcemBdzxLTGIcjPXwPoZNrb",
 	})
 	opts.Grants = append(opts.Grants, Grant{
-		Machine:  "admin",
+		User:     "admin",
 		Role:     "admin",
 		Resource: "infra",
 	})
 }
 
-func TestAPI_CreateIdentity(t *testing.T) {
+func TestDeleteProvider_NoDeleteInternalProvider(t *testing.T) {
```

Comment:
I think this is covered by the `"infra provider can not be deleted"` case in the table test above, no?